src/utils/simplifier.js

```diff
@@ -103,14 +103,52 @@ function simplify(realm, value: Value, isCondition: boolean = false): Value {
         // (x: boolean) || false <=> x
         if (!y.mightNotBeFalse()) return op === "||" ? x : realm.intrinsics.false;
       }
-      if (
-        op === "||" &&
-        y instanceof AbstractValue &&
-        y.kind === "||" &&
-        x.equals(y.args[0]) &&
-        !y.args[1].mightNotBeTrue()
-      )
-        return y;
+      if (op === "||") {
+        if (y instanceof AbstractValue && y.kind === "||") {
+          // x || x || y <=> x || y
+          if (x.equals(y.args[0])) return y;
+          if (x instanceof AbstractValue && x.kind === "!") {
+            // !x0 || x0 || y1 <=> y1
+            if (x.args[0].equals(y.args[0])) return y.args[1];
```

Comment:
This still seems wrong to me. `!x || x || y` should be equivalent to `true`, not `y`.

Reply:
Duh! I'll remove this.